File: java/src/exercises/OOPAssignment.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.Collections;

public class OOPAssignment {
    public static void main() {
        Scanner in = new Scanner(System.in);
        int[] nums = new int[10];

        System.out.println("Enter 10 random numbers.");
        for (int i = 0; i < 10; i++) {
            System.out.printf("#%d: ", i + 1);
            nums[i] = in.nextInt();
        }

        //SortNumbersManual.quickSort(nums, 0, nums.length - 1);
        quickSort(nums, 0, nums.length - 1);
        System.out.println("\nThe sum of 1 to 200 is: " + idk());
        System.out.println("Sorted 10 random numbers: " + Arrays.toString(nums));
    }

    public static void quickSort(int[] arr, int low, int high) {
        if (low < high) {
            int pivotIndex = partition(arr, low, high);
            quickSort(arr, low, pivotIndex - 1);
            quickSort(arr, pivotIndex + 1, high);
        }
    }

    private static int partition(int[] arr, int low, int high) {
        int pivot = arr[high]; 
        int i = (low - 1); 

        for (int j = low; j < high; j++) {
            if (arr[j] > pivot) {
                i++;
                swap(arr, i, j);
            }
        }

        swap(arr, i + 1, high);
        return i + 1;
    }

    private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    static int idk() {
        int sum = 0;
        for (int i = 0; i <= 200; i++) {
            sum += i;
            if (isDivisibleThree(i) && isDivisibleFive(i)) {
                System.out.println(i + " is divisible by 3 and 5");
                continue;
            } else if (isDivisibleThree(i)) {
                System.out.println(i + " is divisible by 3");
                continue;
            } else if (isDivisibleFive(i)) {
                System.out.println(i + " is divisible by 5");
                continue;
            } else {
                System.out.println(i);
                continue;
            }
        }
        return sum;
    }

    static boolean isDivisibleThree(int n) {
        return n % 3 == 0;
    } 

    static boolean isDivisibleFive(int n) {
        return n % 5 == 0;
    } 
}

File: cpp/src/exercises/oop_assignment.cpp
#include <iostream>

class SortNumbers {
    public:
        void quickSort(int arr[], int low, int high) {
            if (low < high) {
                int pivotIndex = partition(arr, low, high);
                
                quickSort(arr, low, pivotIndex - 1);
                quickSort(arr, pivotIndex + 1, high); 
            }
        }

        int partition(int arr[], int low, int high) {
            int pivot = arr[high];
            int i = low - 1;

            for (int j = low; j < high; j++) {
                if (arr[j] > pivot) { 
                    i++;
                    std::swap(arr[i], arr[j]);
                }
            }

            std::swap(arr[i + 1], arr[high]);
            return i + 1;
        }
};

class Numbers {
    public:
        static void idk() {
            int sumOf3 = 0;
            int sumOf5 = 0;
            int sumOf3And5 = 0;
            int sumOf3Or5 = 0;
            int sumOfAll = 0;
            for (int i = 0; i <= 200; i++) {
                if (i % 3 == 0 && i % 5 == 0) {
                    sumOfAll += i;
                    sumOf3And5 += i;
                    std::cout << i << " is divisible by 3 and 5\n";
                    continue;
                } else if (i % 3 == 0) {
                    sumOf3 += i;
                    sumOf3Or5 += i;
                    sumOfAll += i;
                    std::cout << i << " is divisible by 3\n";
                    continue;
                } else if (i % 5 == 0) {
                    sumOf5 += i;
                    sumOf3Or5 += i;
                    sumOfAll += i;
                    std::cout << i << " is divisible by 5\n";
                    continue;
                } 
            }
            std::cout << "\n";
            std::cout << "Sum of 3: " << sumOf3 << "\n";
            std::cout << "Sum of 5: " << sumOf5 << "\n";
            std::cout << "Sum of 3 & 5: " << sumOf3And5 << "\n";
            std::cout << "Sum of 3 + 5: " << sumOf3Or5 << "\n";
            std::cout << "Sum of All: " << sumOfAll << "\n";
        }
};


void number_assignment() {
    Numbers::idk();
}

void sort_10() {
    int nums[10];

    std::cout << "Enter 10 random numbers.\n";
    for (int i = 0; i < 10; i++) {
        std::cout << "#" << i + 1 << ": ";
        std::cin >> nums[i];
    }

    SortNumbers *myObj = new SortNumbers;
    myObj->quickSort(nums, 0, 9);

    std::cout << "\nSorted 10 random numbers: ";
    for (int i = 0; i < 10; i++) {
        std::cout << nums[i] << ", ";
    }
    std::cout << (char) 10;
}

int main() {
    number_assignment();
    return 0;
}